query-engine/query-structure/src/query_arguments.rs

```diff
@@ -70,7 +72,27 @@ impl QueryArguments {
     /// retrieved by the connector or if it requires the query engine to fetch a raw set
     /// of records and perform certain operations itself, in-memory.
     pub fn requires_inmemory_processing(&self) -> bool {
-        self.distinct.is_some() || self.contains_unstable_cursor() || self.contains_null_cursor()
+        self.contains_unstable_cursor() || self.contains_null_cursor() || self.requires_inmemory_distinct()
+    }
+
+    pub fn requires_inmemory_distinct(&self) -> bool {
+        self.has_distinct() && (!self.has_distinct_capability() || self.has_orderby())
+    }
```

Comment:
Check to ensure that we're only doing distinct queries _without_ orderby in db

Reply:
So we're only doing distinct when no order by is present, right? I understand this is a distinct limitation, I just want to make sure this is what you ended up deciding at a product level.